custom_components/ha_creality_ws/coordinator.py

```
@@ -378,15 +387,86 @@ async def _handle_message(self, payload: dict[str, Any]) -> None:
+        _LOGGER.debug(
+            "Notification baseline captured: file=%s progress=%s completed=%s "
+            "err=%s runout=%s near_end=%s",
+            self._last_filename,
+            prog_val,
+            self._notified_completed,
+            self._last_error_code,
+            self._notified_filament_runout,
+            self._notified_minutes_to_end,
+        )
+
     async def _check_notifications(self, _payload: dict[str, Any]):
```

Ruff (0.16.1)

[warning] Too many branches (22 > 12)

(PLR0912)